cms/djangoapps/contentstore/views/assets.py

```diff
@@ -65,7 +65,7 @@ def assets_handler(request, course_key_string=None, asset_key_string=None):
         else:
             asset_key = AssetKey.from_string(asset_key_string) if asset_key_string else None
             return _update_asset(request, course_key, asset_key)
-    elif request.method == 'GET':  # assume html
+    elif request.method == 'GET' or request.method == 'POST':  # assume html
```

Comment:
I'm concerned about the different behavior of json response vs. html. If the response_format is json, GET will return all the assets, and POST assumes an individual asset should be updated.

However if response_format is html, GET or POST will return a list of assets (possibly filtered if request. method is POST).

I think we need more parallel behavior, so that it is possible to return a filtered asset list when response_format is JSON. Also, POST doesn't feel like the right verb here-- it really should be a GET with query parameters. See https://openedx.atlassian.net/wiki/display/AC/edX+REST+API+Conventions for filtering examples.

This means that you won't be able to use a form-- you will need to get the filter_criteria from the input when Filter is pressed, and create the GET url.